src/MongoDB.Driver/Encryption/AutoEncryptionLibMongoController.cs

```diff
@@ -223,5 +227,39 @@ private void RestoreDbNodeInResponse(BsonDocument request, BsonDocument response
                 }
             }
         }
+
+        private void WaitForMongocryptdReady()
+        {
+            using (CancellationTokenSource cancellationTokenSource = new CancellationTokenSource(TimeSpan.FromMinutes(1)))
+            {
+                while (!cancellationTokenSource.IsCancellationRequested)
```

Comment:
This usage of a CancellationTokenSource is a bit unusual because we never actually use the CancellationToken itself for anything.

Consider using a plain StopWatch instead:

    var stopwatch = Stopwatch.StartNew();
    while (stopwatch.Elapsed < TimeSpan.FromMinutes(1))
    {
        // ...
    }